tools/mo/openvino/tools/mo/back/preprocessing.py

```diff
@@ -332,6 +345,10 @@ def apply_preprocessing(ov_function: Function, argv: argparse.Namespace):
     # On return, mean_scale_values is a dictionary with input names as key and mean/scale pair as value
     # {'inputName': {'mean': [1., 2., 3.], 'scale': [2.]}}
 
+    # The MO configuration could have modified the orignal model before preprocessing started and thus
+    # the mean_scale_values might contain keys that don't match the modified model's inputs
+    mean_scale_values = prune_obsolete_entries(mean_scale_values, ov_function.inputs)
```

Comment:
How legacy MO behaves for this case? In general, if user passes incorrect name for mean/scale it shall throw "Input with name was not found" error, it shall not assume that input might have been deleted by cut operations.

Reply:
I've removed those changes because the preprocessing stage needs to be updated in general. Parsing of keys in `mean_scale_values` should be added because they might represent original values passed to the mo.py script. This is a separate activity and I can create a new story for that.